src/coreclr/jit/targetwasm.h

```diff
@@ -0,0 +1,284 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#pragma once
+
+#if !defined(TARGET_WASM)
+#error The file should not be included for this platform.
+#endif
+
+// clang-format off
+#define CPU_LOAD_STORE_ARCH      1
+#define CPU_HAS_FP_SUPPORT       1
+#define CPU_HAS_BYTE_REGS        0
+
+// Currently we don't pass any arguments on the linear memory stack so FEATURE_FIXED_OUT_ARGS is not needed.
+#define FEATURE_FIXED_OUT_ARGS   0       // Preallocate the outgoing arg area in the prolog
+#define FEATURE_STRUCTPROMOTE    1       // JIT Optimization to promote fields of structs into registers
+#define FEATURE_MULTIREG_STRUCT_PROMOTE 1  // True when we want to promote fields of a multireg struct into registers
+#define FEATURE_FASTTAILCALL     0       // Tail calls made as epilog+jmp
+#define FEATURE_TAILCALL_OPT     0       // opportunistic Tail calls (i.e. without ".tail" prefix) made as fast tail calls.
+#define FEATURE_IMPLICIT_BYREFS       1  // Support for struct parameters passed via pointers to shadow copies
+#define FEATURE_MULTIREG_ARGS_OR_RET  0  // Support for passing and/or returning single values in more than one register
+#define FEATURE_MULTIREG_ARGS         0  // Support for passing a single argument in more than one register
+#define FEATURE_MULTIREG_RET          0  // Support for returning a single value in more than one register
+#define MAX_PASS_SINGLEREG_BYTES      8  // Maximum size of a struct passed in a single register (long/double).
+#define MAX_PASS_MULTIREG_BYTES       0  // Maximum size of a struct that could be passed in more than one register
+#define MAX_RET_MULTIREG_BYTES        0  // Maximum size of a struct that could be returned in more than one register (Max is an HFA or 2 doubles)
+#define MAX_ARG_REG_COUNT             1  // Maximum registers used to pass a single argument in multiple registers.
+#define MAX_RET_REG_COUNT             1  // Maximum registers used to return a value.
+#define MAX_MULTIREG_COUNT            2  // Maximum number of registers defined by a single instruction (including calls).
+                                         // This is also the maximum number of registers for a MultiReg node.
+
+#define NOGC_WRITE_BARRIERS      0       // No specialized WriteBarrier JIT Helpers
+#define USER_ARGS_COME_LAST      1
```

Comment:
It might be beneficial for performance for them to come first, depending on how VMs allocate wasm args to registers (whether it's the first N args in the signature, or they do some sort of smarter analysis). But that is at most a WASM-TODO imo.

Reply:
> don't want to burn one of them on the generic context.

Nit: I believe that this calling convention was chosen to make implementation of instantiating stubs via hand-emitted assembly easier on Windows x86. I do not think there was a strong performance reason for it.